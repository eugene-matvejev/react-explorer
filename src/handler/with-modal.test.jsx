import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import withModal from './with-modal';

const { WrappedComponent: WithModal } = withModal(() => <div className="{{wrappedComponent}}" />);

configure({ adapter: new Adapter() });

describe('<WithModal/> as HOC', () => {
    const props = {
        location: {},
        history: {
            goBack: jest.fn(),
        },
        'data-cy': '{{data-cy}}', //should be passed "as is",
        modals: [
            {
                path: ['/explore/:id'],
                component: (props) => <div className="{{modal}}" />,
            },
        ],
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<WithModal {...props} />);

            expect(c).toMatchSnapshot();
        });

        it('with location.state.isModal as true', () => {
            const c = shallow(<WithModal {...props} location={{ state: { isModal: true } }} />);

            expect(c).toMatchSnapshot();
        });
    });
});
