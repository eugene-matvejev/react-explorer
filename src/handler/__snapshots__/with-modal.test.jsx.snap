// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithModal/> as HOC render with default/required props 1`] = `
<Fragment>
  <Component
    data-cy="{{data-cy}}"
    history={
      Object {
        "goBack": [MockFunction],
      }
    }
  />
</Fragment>
`;

exports[`<WithModal/> as HOC render with location.state.isModal as true 1`] = `
<Fragment>
  <Component
    data-cy="{{data-cy}}"
    history={
      Object {
        "goBack": [MockFunction],
      }
    }
  />
  <_default
    onClose={[MockFunction]}
  >
    <Switch>
      <Route
        component={[Function]}
        data-cy="{{data-cy}}"
        history={
          Object {
            "goBack": [MockFunction],
          }
        }
        key="0"
        path={
          Array [
            "/explore/:id",
          ]
        }
      />
    </Switch>
  </_default>
</Fragment>
`;
